src/core/mac/sub_mac.cpp

```diff
@@ -1029,6 +1029,145 @@ void SubMac::StartTimerAt(Time aStartTime, uint32_t aDelayUs)
+ *             ^       ^
+ *             |       |
+ *             | RequestSleep(Csl)
+ *     RequestReceive(Csl)
+ *
+ * WED   -----------++++++++----------------++++++++----------------++++++++----------
+ *                  ^       ^
+ *                  |       |
+ *                  | RequestSleep(Wed)
+ *         RequestReceive(Wed)
+ *
+ * Radio ------========+++++-------========-++++++++---========-----+++++++========---
+ *             ^       ^    ^
+ *             |       |    |
+ *             |       | Radio::Sleep()
+ *             |  Radio::Receive(WedCh)
+ *      Radio::Receive(CslCh)
+ */
+void SubMac::RequestSleep(Requester aRequester)
+{
+    if (IsCslEnabled() && mIsWedEnabled)
+    {
+        if ((aRequester == kRequesterCsl) && mIsWedSampling)
+        {
+            IgnoreError(Get<Radio>().Receive(mWakeupChannel));
+        }
+        else if ((aRequester == kRequesterWed) && mIsCslSampling)
+        {
+            // When the CSL is in the receive state and the WED enters the sleep state, the radio should have entered
+            // receive state using CSL channel. Do nothing here.
+        }
+        else
+        {
+            IgnoreError(Get<Radio>().Sleep());
+        }
+    }
+    else if (IsCslEnabled() || mIsWedEnabled)
+    {
+        IgnoreError(Get<Radio>().Sleep());
+    }
+}
+
+void SubMac::RequestReceive(Requester aRequester)
+{
+    if (aRequester == kRequesterCsl)
+    {
+        IgnoreError(Get<Radio>().Receive(mCslChannel));
+    }
+    else if ((aRequester == kRequesterWed) && ((IsCslEnabled() && !mIsCslSampling) || !IsCslEnabled()))
+    {
+        IgnoreError(Get<Radio>().Receive(mWakeupChannel));
+    }
+}
+
+bool SubMac::RequestSample(void)
+{
+    bool ret = false;
+
+    if (mIsCslSampling && IsCslEnabled())
+    {
+        IgnoreError(Get<Radio>().Receive(mCslChannel));
+        ret = true;
+    }
+    else if (mIsWedSampling && mIsWedEnabled)
+    {
+        IgnoreError(Get<Radio>().Receive(mWakeupChannel));
+        ret = true;
+    }
+
+    return ret;
+}
+
+#elif OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+bool SubMac::IsRadioSampleEnabled(void) const { return IsCslEnabled(); }
+void SubMac::RequestSleep(Requester) { IgnoreError(Get<Radio>().Sleep()); }
+void SubMac::RequestReceive(Requester) { IgnoreError(Get<Radio>().Receive(mCslChannel)); }
+
+bool SubMac::RequestSample(void)
+{
+    if (mIsCslSampling)
+    {
+        IgnoreError(Get<Radio>().Receive(mCslChannel));
+    }
+
+    return mIsCslSampling;
+}
+#elif OPENTHREAD_CONFIG_WAKEUP_END_DEVICE_ENABLE
+bool SubMac::IsRadioSampleEnabled(void) const { return mIsWedEnabled; }
+void SubMac::RequestSleep(Requester) { IgnoreError(Get<Radio>().Sleep()); }
+void SubMac::RequestReceive(Requester) { IgnoreError(Get<Radio>().Receive(mWakeupChannel)); }
+
+bool SubMac::RequestSample(void)
+{
+    if (mIsWedSampling)
+    {
+        IgnoreError(Get<Radio>().Receive(mWakeupChannel));
+    }
+
+    return mIsWedSampling;
+}
```

Comment:
Suggest to add some seperator lines and doc to indicate this block is to implement the functions for WED only case. Add also do this to the other 2 cases.